projects/forms/src/lib/directives/errors/errors.directive.ts

```diff
@@ -0,0 +1,240 @@
+		])
+			.pipe(
+				tap(([, touched]) => {
+					// Iben: Check whether we should show the error based on the provided config
+					const shouldShow =
+						this.abstractControl.invalid &&
+						(this.config.showWhen === 'touched' ? touched : this.abstractControl.dirty);
+
+					// Iben: Show the error based on whether or not a component was provided
+					if (!this.config.component) {
+						this.handleNoComponentFlow(shouldShow);
+					} else {
+						this.handleComponentRender(shouldShow);
+					}
+				}),
+				takeUntil(this.onDestroy$)
+			)
+			.subscribe();
+	}
+
+	/**
+	 * Renders a provided custom component underneath the input component
+	 *
+	 * @param shouldShow - Whether the error should be shown
+	 */
+	private handleComponentRender(shouldShow: boolean) {
+		// Iben: If the error should not be shown, we check if there's already an error component and destroy it if needed
+		if (!shouldShow) {
+			if (this.errorComponent) {
+				this.componentRef.destroy();
+				this.componentRef = undefined;
+				this.errorComponent = undefined;
+			}
+
+			return;
+		}
+
+		// Iben: If there's no error component yet, we add one to the view
+		if (!this.errorComponent) {
+			this.componentRef = this.viewContainer.createComponent<NgxFormsErrorAbstractComponent>(
+				this.config.component
+			);
+			this.errorComponent = this.componentRef.instance;
+		}
+
+		// Iben: Set the data of the error component
+		const { errors, errorKeys, data } = this.getErrors(this.abstractControl.errors);
+
+		this.errorComponent.errors = errors;
+		this.errorComponent.errorKeys = errorKeys;
+		this.errorComponent.data = data;
+	}
+
+	/**
+	 * Renders a p tag underneath the input component when no custom component was provided
+	 *
+	 * @param shouldShow - Whether the error should be shown
+	 */
+	private handleNoComponentFlow(shouldShow: boolean) {
+		// Iben: We remove the current errors so that we always have a new element to work with
+		if (this.errorsElement) {
+			this.renderer.removeChild(this.elementRef.nativeElement.parentNode, this.errorsElement);
+
+			this.errorsElement = null;
+		}
+
+		// Iben: Early exit in case there's no error to show
+		if (!shouldShow) {
+			return;
+		}
+
+		// Iben: Create a new error paragraph
+		this.errorsElement = this.renderer.createElement('p');
+		this.renderer.setAttribute(this.errorsElement, 'class', 'ngx-forms-error');
+
+		// Iben: Set the errors based on the keys
+		this.renderer.setProperty(
+			this.errorsElement,
+			'innerHTML',
+			this.getErrors(this.abstractControl.errors).errors.join(', ')
+		);
+
+		// Iben: insert the paragraph underneath the input component
+		this.renderer.insertBefore(
+			this.elementRef.nativeElement.parentNode,
+			this.errorsElement,
+			this.renderer.nextSibling(this.elementRef.nativeElement)
+		);
+	}
+
+	/**
+	 * Returns the errors based on the provided settings
+	 *
+	 * @param data - The error data we wish to use
+	 */
+	private getErrors(data: ValidationErrors): {
```

Comment:
Hier nog even afchecken of data een object is denk ik. Als die nu gaat proberen van `Object.keys()` te doen en data is geen object dan breekt de code. Slim chance maar beter wat extra veiligheid inbouwen.

Reply:
Eigenlijk geen slim chance at all; want dat kan inderdaad null zijn 🙈 Goed gezien!